### ISSUE TITLE
TC39 top-level-await is now stage 4 -- please handle as expected

### ISSUE THREAD

**Issue by JESii** [eslint/eslint#15969]
### Syntax name

top-level-await 

### Syntax proposal URL

https://github.com/tc39/proposals/blob/main/finished-proposals.md

### Example code

```js
await processLineByLine(inputFile)
  /* eslint-disable-next-line no-unused-vars */
  .then((result) => {
    console.log("That's all folks!");
    process.exit(0);
  })
  .catch((err) => {
    console.log(err);
    process.exit(1);
  });
```

### Implementation Checklist

- [ ] Ecma262 update: 
- [ ] ESTree update: 
- [ ] Acorn update: 
- [ ] `eslint-visitor-keys` update: 
- [ ] `espree` update: 
- [ ] `eslint-scope` update: 
- [ ] `eslint` update: 

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by ljharb** [comment]
It should work fine if you have ecmaVersion set to 2022, and the file is an ESM module.

**Comment by mdjermanovic** [comment]
ESLint supports top-level await syntax since [v8.0.0](https://eslint.org/blog/2021/10/eslint-v8.0.0-released). You'll just need to set `parserOptions.ecmaVersion` to `2022` or `"latest"`.

```json
{
    "parserOptions": {
        "sourceType": "module",
        "ecmaVersion": 2022
    }
}
```

Here's a [demo example](https://eslint.org/demo#eyJ0ZXh0IjoiYXdhaXQgcHJvY2Vzc0xpbmVCeUxpbmUoaW5wdXRGaWxlKVxuICAvKiBlc2xpbnQtZGlzYWJsZS1uZXh0LWxpbmUgbm8tdW51c2VkLXZhcnMgKi9cbiAgLnRoZW4oKHJlc3VsdCkgPT4ge1xuICAgIGNvbnNvbGUubG9nKFwiVGhhdCdzIGFsbCBmb2xrcyFcIik7XG4gICAgcHJvY2Vzcy5leGl0KDApO1xuICB9KVxuICAuY2F0Y2goKGVycikgPT4ge1xuICAgIGNvbnNvbGUubG9nKGVycik7XG4gICAgcHJvY2Vzcy5leGl0KDEpO1xuICB9KTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEzLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19).

**Comment by JESii** [comment]
Thanks much @mdjermanovic ; lots of plugin updates to get v8 to work, but now on v8.17.0.
